partiql-planner/src/test/kotlin/org/partiql/planner/internal/typer/PlanTyperTestsPorted.kt

```diff
@@ -289,6 +293,37 @@ class PlanTyperTestsPorted {
         @JvmStatic
         fun structs() = listOf<TestCase>()
 
+        @JvmStatic
+        fun extendedCastTests() = listOf<TestCase>(
+            SuccessTestCase(
+                name = "CAST INT4 TO INT2",
+                catalog = CATALOG_AWS,
+                query = "CAST((SELECT t.id FROM ddb.pets AS t) AS INT2)",
+                expected = INT2
+            ),
+            SuccessTestCase(
+                name = "CAST BOOL TO INT2",
+                catalog = CATALOG_AWS,
+                query = "CAST(true AS INT2)",
+                expected = INT2
+            ),
+            SuccessTestCase(
+                name = "CAST INT2 TO INT4",
+                catalog = CATALOG_AWS,
+                query = "CAST((SELECT t.tag FROM ddb.pets AS t) AS INT4)",
+                expected = INT4
+            ),
+            SuccessTestCase(
+                name = "CAST INT8 literal to INT2",
+                query = "CAST(343434534534 AS INT2)",
+                expected = INT2
+            ),
+            ErrorTestCase(
+                name = "CAST STRING to INT4",
+                query = "CAST((SELECT t.breed FROM ddb.pets AS t) AS INT2)",
+            ),
```

Comment:
`CAST(<string> TO <int>)` should not be a type-check error, but it _may_ be a runtime error.

- I checked postgres and they allow `CAST('1' AS INT)`
- Existing [PLK behavior allows this](https://github.com/partiql/partiql-lang-kotlin/wiki/Functions#cast----since-v010) and per recent discussions we should do our best to maintain current functional behavior.


Additionally, the test name has `INT4` whereas the query uses `INT2`.

Reply:
Makes sense, will fix.